client/src/components/CheckboxFilters/index.js

```diff
@@ -3,43 +3,68 @@ import CheckboxItem from "./CheckboxItem";
 import Ajax from "../../services/Ajax";
 import './styles.less';
 
-const CheckboxFilter = ({filters, clickCheckbox}) => {
+const CheckboxFilter = ({clickCheckbox, query}) => {
 
+    const [filters, setFilters] = useState([]);
     const [catalog, setCatalog] = useState([]);
-    const allTypes = filters.map(item => {
-        return item.type
-    })
-    const uniqTypes = Array.from(new Set(allTypes));
 
     useEffect(() => {
         async function fetch() {
-            const result = await Ajax.get('/catalog');
+            const result = (await Ajax.get('/catalog'))
+                .filter(item => item.level === "2")
             setCatalog(result);
         }
 
         fetch()
     }, []);
+    useEffect(() => {
+        async function fetch() {
+            const result = (await Ajax.get('/filters'))
+                .map(item => item.type)
+                .filter((v, i, a) => a.indexOf(v) === i)
+                .map(async type => {
+                    const result = await Ajax.get(`/filters/${type}`);
+                    return {
+                        type,
+                        values: result
+                    }
+                });
+
+            setFilters(await Promise.all(result));
+        }
+
+        fetch()
+    }, []);
 
-    const filteredCatalog = catalog.filter(item => item.level === "2");
-    console.log('catalog--->', filteredCatalog);
     return (
-        <div className='checkbox-container' onClick={clickCheckbox}>
+        <div className='checkbox-container'>
             <div className='checkbox-group checkbox-catalog-group'>
                 <p className='checkbox-group__name'>Catalog</p>
                 {
-                    filteredCatalog.map(item =>
-                        <div className='checkbox-group__item' key={item.name}>
-                            <input className='item-filter' data-type={'catalog'} type="checkbox" id={item.name} name={item.name}/>
-                            <label className='checkbox-label' htmlFor={item.name}>{item.name}</label>
-                        </div>
+                    catalog.map(category =>
+                        <CheckboxItem key={category.id}
+                                      type={'categories'}
+                                      checked={query.categories?.includes(category.id) || false}
```

Comment:
We agreed with @KosSid to keep using categories name instead of id for filters. This must, the line below) and probably something else must be changed to name instead of id.